### PR TITLE
Add shadows I + P datataking module. 5/n

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Adds the final experiment: I + P using a single copy + shadows.

### CODE REVIEW

**Review by mpharrigan — APPROVED** [review]
just gave a cursory look, but LGTM

File: recirq/qml_lfe/learn_states_c_test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import os
+import pytest
+import tempfile
+import numpy as np
+from . import learn_states_c
+
+
+def _predict_exp(data, pauli_list, two_copy):
```
Comment (dstrain115) [inline recirq/qml_lfe/learn_states_c_test.py:8]: Might be worth a comment.  Also, is this the same as _predict_exp from the other tests?

File: recirq/qml_lfe/circuit_blocks.py [hunk 2]
```diff
@@ -190,3 +191,51 @@ def inv_z_basis_gate(pauli: str) -> cirq.Gate:
             axis_phase_exponent=-0.5, x_exponent=0.5, z_exponent=-0.5
         )
     raise ValueError("Invalid Pauli.")
+
+
+def create_randomized_sweeps(
+    hidden_p: str,
+    symbols: Dict[str, int],
+    num_reps: int,
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply recirq/qml_lfe/circuit_blocks.py]: Ahhh good call. I changed this over to `n_shots` everywhere. In our case we take just one shot per sweep so the number of sweeps and number of shots are the same number.

File: recirq/qml_lfe/circuit_blocks.py [hunk 3]
```diff
@@ -190,3 +191,51 @@ def inv_z_basis_gate(pauli: str) -> cirq.Gate:
...
+    for i, pauli in enumerate(hidden_p):
+        if pauli != "I":
+            last_i = i
+
+    sign_p = -1 if rand_state.random() < 0.5 else 1
+    all_sweeps = []
+    for _ in range(num_reps):
+        current_sweep = dict()
+        for twocopy in [0, 1]:
+            parity = sign_p * (1 if rand_state.random() <= 0.95 else -1)
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply recirq/qml_lfe/circuit_blocks.py]: Oops, fixed.

File: recirq/qml_lfe/learn_states_c.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Draw I + P pauli product data using shadow strategy.
+style circuits.
+
+learn_states_c.py --n=6 --n_paulis=20 --batch_size=500 --n_shots=1000
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply recirq/qml_lfe/learn_states_c.py]: Changed.

File: recirq/qml_lfe/learn_states_c.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+Will create paulistring circuits all on 6 qubits.
+Once the circuits are generated, batch_size sweeps and a single circuit at
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply recirq/qml_lfe/learn_states_c.py]: Tightened things up a bit, think it reads OK now, WDYT ?

File: recirq/qml_lfe/learn_states_c.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+a time will be sent will be sent for simulation/execution, drawing n_shots
+total samples from each one using `run_sweep` in Cirq. By default the
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply recirq/qml_lfe/learn_states_c.py]: Added.

File: recirq/qml_lfe/learn_states_c_test.py (hunk 1 above)
Comment (MichaelBroughton) [reply recirq/qml_lfe/learn_states_c_test.py]: Added. It's the same.

**Review by dstrain115 — APPROVED** [review]
LGTM from me.

**Pull request merged**